### PR TITLE
Compare versions across REST API

### LINKED ISSUE #544: Inform user that a new version of the CLI is available
Nice feature to have.

CLI should periodically check (once a day when a cl command is run?) if there are updates available, then inform the user once.

### PR DESCRIPTION
**Pull request by kashizui**
Notify CLI user when a new version is available.

Fixes #544

@percyliang

### CODE REVIEW

File: codalab/lib/codalab_manager.py [hunk 1]
```diff
@@ -605,6 +612,25 @@ def set_current_worksheet_uuid(self, address, worksheet_uuid):
             if 'worksheet_uuid' in session: del session['worksheet_uuid']
         self.save_state()
 
+    def check_version(self, server_version):
+        # Enforce checking version at most once every 24 hours
+        now = datetime.datetime.now()
+        last_check_dt = formatting.parse_datetime(self.state.get('last_check_version', formatting.datetime_str(datetime.datetime.utcfromtimestamp(0))))
```
Comment (percyliang) [inline codalab/lib/codalab_manager.py]: Replace last_check_version with something like last_check_version_datetime for clarity

File: codalab/lib/codalab_manager.py [hunk 2]
```diff
@@ -605,6 +612,25 @@ def set_current_worksheet_uuid(self, address, worksheet_uuid):
...
+        if (now - last_check_dt) < datetime.timedelta(days=1):
+            return
+        self.state['last_check_version'] = formatting.datetime_str(now)
+        self.save_state()
+
+        # Print notice if server version is newer
+        if server_version.split('.') > CODALAB_VERSION.split('.'):
```
Comment (percyliang) [inline codalab/lib/codalab_manager.py]: Don't you need to convert to integers? Otherwise 11 will come before 9....

**Review by percyliang — APPROVED** [review]
Cool!  LGTM modulo comments

**Comment by percyliang** [comment]
Tests are failing...

File: codalab/lib/codalab_manager.py (hunk 2 above)
Comment (kashizui) [reply codalab/lib/codalab_manager.py]: Yikes, good catch

**Comment by kashizui** [comment]
Added support for XMLRPC as well.

**Pull request merged**